report/report.tex

```diff
@@ -37,6 +66,50 @@ \section{Methodology}
 	\subsection{Planning}
 \section{Results}
 	\subsection{Achievements}
+	
+	
+	
+	\subsubsection{Semantics of operators}	
+	In Hanus, a user is able to define their own types and operators. These operators can be defined for their own types or for types that are already defined in Haskell. In Hanus, there are already a number of operators and types defined that users can use. 
+	\paragraph{Definition of arbitrary types}
+	In a reversible language, any variable should have a default value with which a variable is initialised before the \textit{main} procedure is run. The original \textit{Janus} specification only allowed for integers. For integers the default value of 0 is chosen. In Hanus, for any number of the \textit{Haskell Num} class, the default is also 0. There are also a number of other types predefined in Hanus. Their defaults are implemented as follows:
+	\begin{itemize}
+		\item Bool: \textit{False}
+		\item $\lbrack\alpha\rbrack$ : \textit{[]}
+		\item Map $\alpha$ $\beta$: \textit{Data.Map.empty}
+		\item Maybe $\alpha$: \textit{Nothing}
+	\end{itemize}
+	Any user that wants to define their own type, has to implement an instance of \textit{DefaultValue} for the type they want to use as a \textit{Hanus} variable. It is possible to implement a \textit{DefaultValue} for a type without implementing any operators for that type. A variable of this type can not be changed in this case, except if the type is a subtype of another type. An example of this would be the class \textit{Real}. Because any \textit{Real} in Haskell has to be also a \textit{Num}, all operators defined on \textit{Num} could also be used on a \textit{Real}.
```

Comment:
> subtype

Haskell has no subtypes, you are referring to operators that can be used since they were defined on a type class. That means that they thus do have operators defined, so I'd remove the part starting at "except if the type"